joern-cli/frontends/php2cpg/src/main/scala/io/joern/php2cpg/utils/Scope.scala

```diff
@@ -30,109 +35,113 @@ class Scope(summary: Map[String, Seq[SymbolSummary]] = Map.empty)(implicit nextC
-        logger.warn(s"pushNewScope called with invalid node $invalid. Ignoring!")
+        typeDecl
+      case namespace: NamespaceScope => namespace
     }
+
+    super.pushNewScope(mappedNode)
   }
 
   def surroundingAstLabel: Option[String] = stack.collectFirst {
-    case ScopeElement(_: NamespaceLikeScope, _)                 => NodeTypes.NAMESPACE_BLOCK
-    case ScopeElement(PhpScopeElement(x: NewNamespaceBlock), _) => NodeTypes.NAMESPACE_BLOCK
-    case ScopeElement(PhpScopeElement(x: NewTypeDecl), _)       => NodeTypes.TYPE_DECL
-    case ScopeElement(PhpScopeElement(x: NewMethod), _)         => NodeTypes.METHOD
+    case ScopeElement(_: NamespaceLikeScope, _) => NodeTypes.NAMESPACE_BLOCK
+    case ScopeElement(_: TypeLikeScope, _)      => NodeTypes.TYPE_DECL
+    case ScopeElement(_: MethodLikeScope, _)    => NodeTypes.METHOD
   }
 
   def surroundingScopeFullName: Option[String] = stack.collectFirst {
-    case ScopeElement(x: NamespaceLikeScope, _)                 => x.fullName
-    case ScopeElement(PhpScopeElement(x: NewTypeDecl), _)       => x.fullName
-    case ScopeElement(PhpScopeElement(x: NewMethod), _)         => x.fullName
-    case ScopeElement(PhpScopeElement(x: NewNamespaceBlock), _) => x.fullName
+    case ScopeElement(x: NamespaceLikeScope, _) => x.fullName
+    case ScopeElement(x: TypeLikeScope, _)      => x.fullName
+    case ScopeElement(x: MethodLikeScope, _)    => x.fullName
   }
 
-  override def popScope(): Option[PhpScopeElement] = {
+  override def popScope(): Option[TypedScopeElement] = {
     val scopeNode = super.popScope()
 
-    scopeNode.map(_.node) match {
-      case Some(_: NewTypeDecl) =>
+    scopeNode match {
+      case Some(_: TypeScope) =>
         // TODO This is unsafe to catch errors for now
         constAndStaticInits = constAndStaticInits.tail
         fieldInits = fieldInits.tail
-
-      case _ => // Nothing to do here
+      case _ => // nothing to do here
     }
 
     scopeNode
   }
 
+  private def getNextClassTmp: String = {
+    val returnString = s"anon-class-${tmpClassCounter}"
+    tmpClassCounter += 1
+
+    returnString
+  }
+
+  private def getNextVarTmp: String = {
+    val returnString = s"tmp-${tmpVarCounter}"
+    tmpVarCounter += 1
+
+    returnString
+  }
+
   def getNewClassTmp: String = {
     stack.headOption match {
-      case Some(node) =>
-        s"${this.surroundingScopeFullName.getOrElse("<global>")}.${node.scopeNode.getNextClassTmp}"
-      case None =>
+      case Some(_ @ScopeElement(namespace: NamespaceScope, _)) =>
+        s"${this.surroundingScopeFullName.getOrElse("<global>")}.${namespace.getNextClassTmp}"
+      case Some(_ @ScopeElement(typeScope: TypeScope, _)) =>
+        s"${this.surroundingScopeFullName.getOrElse("<global>")}.${typeScope.getNextClassTmp}"
+      case Some(_ @ScopeElement(methodScope: MethodScope, _)) =>
+        s"${this.surroundingScopeFullName.getOrElse("<global>")}.${methodScope.getNextClassTmp}"
+      case _ =>
```

Comment:
Here and all other places where you use `_@XXX()` the `_@` is unnecessary. Please remove it.